app/models/concerns/powers/church_powers.rb

```diff
@@ -36,16 +36,26 @@ def assignable_church_ministries
     end
 
     def assignable_church_user_created_church_ministries
-      # assigment.ministry_id is going to be the id the user is trying to create a church on
-      [assignment.ministry] if assignment.present? && !assignment.blocked?
+      # inherited_assignment.ministry_id is going to be the id the user is trying to create a church on
+      [fallback_assignment.ministry] unless blocked?
     end
 
     def visiable_local_churches_security
-      if assignment.blank? || assignment.blocked_role?
+      if blocked?
         Church.securities['registered_public_church']
-      else
+      elsif assignment.blank? || assignment.self_assigned?
```

Comment:
I added the local_private option back on the filter because they still need to be sent to users somehow.